src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -1345,15 +1344,14 @@ internal static bool IsSpanChar(this TypeSymbol type)
             && arguments[0].SpecialType == SpecialType.System_Char;
         }
 
-        internal static bool IsReadOnlySpanChar(this TypeSymbol type)
+        internal static bool IsReadOnlySpanChar(this TypeSymbol? type)
```

Comment:
>internal static bool IsReadOnlySpanChar(this TypeSymbol? type) [](http://example.com/codeflow?start=8&length=62)

This looks like a style-only change. If so, it should be reverted.

Reply:
To clarify, I am referring to the change in the body, not the nullability change for parameter's type.

Reply:
> I had to touch this helper because of nullability anyway, 

This doesn't mean that you have a carte blanche to change the body just because, according to your liking,  it would look "better".

> why not make it a bit better then?

Better is a subjective term. What is better for one person, is worse for another. We, as a team, decided that we do not want to have arguments about style, and that we are not going to welcome style only changes. I believe you are well aware of that position.

Also, every change has non zero cost and non zero risk. Every change might have unintended consequences. It has to be reviewed by several people on the team, etc. If there is no need to change code's behavior, we prefer to not touch the code.

If you think that your change brings some improvement that is beyond style (results in faster execution, etc.), feel free to submit a separate PR and make you case, rather than trying to sneak it in under umbellar of an unrelated change. The way it looks to me this change has no relation to the stated goal of the PR. No test is going to fail when changes in the body are reverted.

Reply:
I'm sorry to say that, but the wording of your response seems unnecessarily passively aggressive towards me. Even though I know the "no style changes" policy, the concept of a "style-only change" is not a precisely defined scientific term, thus its interpretation may vary slightly in the understanding of different people. E.g. I might apply refactoring like that if it simplifies what was previously written by making less code do the same amount of work and I change other code nearby. In such cases the added value IMO seems to be in improved readability, e.g. in this specific case a pattern was easier to reason about than an `&&` expression with index access in it and so on. I _fully agree_ that it is subjectiveness territory and _you are right_ when pointing this out. Yet this doesn't give you a right to accuse me in "trying to sneak it in under umbellar of an unrelated change" or something. You are a team member, so you "by definition" have the last word in any discussion of such sort after all, you don't need to accuse external contributors in low-qualify submissions to force them do what you request. Having said that, I'm reverting the change, thanks for an explanation

Reply:
> the wording of your response seems unnecessarily passively aggressive towards me. 

I apologize for that. That came across wrong and was not my intent. I will definitely try to avoid this in the future.